### PR TITLE
Restructure docs for easier consumption

### PR DESCRIPTION
**Pull request by crwr45**
Adjustments to the documentation, mostly to get a quick start in the README, and begin to make a developer intro.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,461 +1,74 @@
...
-reference, meaning no need to specify them in the resource templates. The fields included will not be showed in the
-output even if they are specified in the resource templates. Omitted fields can be nested therefore each field is
-represented by a list of strings. As can be seen in the example below.
-
-Example for metadata.yaml:
-
-```yaml
-
-Parts:
-  - name: guestbook
```
Comment (pixelsoccupied) [inline README.md:288]: In readme can we have yaml block at shows all the possible metadata.yaml keys with a comment for explanation? And also a directory tree with example references?

Comment (crwr45) [reply README.md:288]: I've added an explained example, but in the interests of brevity and ease of maintenance I've not included every possible key/value or a directory tree as those are covered elsewhere in the repo.
I opted instead for giving an overview and referring the reader to the reference config guide, and the test data for more information and examples. Does that cover what you wanted to see in the main readme?

Comment (pixelsoccupied) [reply README.md:288]: Sounds good!

**Comment by pixelsoccupied** [comment]
merging this for now! 
/approve

**Pull request merged**